test/runtests.jl

```diff
@@ -17,60 +22,308 @@ import Unitful
     @test_throws AssertionError AbstractFFTs.brfft_output_size(A1, 10, 2)
 end
 
-mutable struct TestPlan{T} <: Plan{T}
+mutable struct TestPlan{T,N} <: Plan{T}
```

Comment:
Probably this `TestPlan` and associated DFT implementation should go in a separate file.

Reply:
Yes, I agree. I thought it might be easier to review if the test plans and e.g. the ChainRules tests are not moved to separate files in the initial PR but I'll extract them in an additional commit.

Reply:
I moved the test plans to a separate file.